src/api.ts

```diff
@@ -155,6 +167,53 @@ export function getPortForContainer(hash: CommitHash): number|boolean {
 		: false;
 }
 
+var repoClonePromises: Map<string, Promise<git.Repository> > = new Map();
```

Comment:
do you not want these in `APIState`?

Reply:
what's the benefit?

Reply:
I moved a bunch of things into one state atom to make it easier for runtime introspection. we don't have the display yet, but shoving all the state together is easier for my poor brain to comprehend